### PR TITLE
Reduced test framework dependencies

### PR DESCRIPTION
**Pull request by kshakir**
- Bypass aging Scala Mockito wrappers with new custom wrapper, mostly
- For hard to port Mockito wrappers instead use Java API
- Only run ScalaTest, not Specs2 nor ScalaCheck
- Ported generator specs to scalatestplus-scalacheck
- Unignored and fixed up tests taggeed with PostWomTest
- Turned off ScalaTest HTML reports & removed out of date rendering dependencies

### CODE REVIEW

**Review by mcovarr — APPROVED** [review]

File: cloud-nio/cloud-nio-impl-ftp/src/test/scala/cloud/nio/impl/ftp/FtpCloudNioFileSystemProviderSpec.scala [hunk 1]
```diff
@@ -42,30 +43,33 @@ class FtpCloudNioFileSystemProviderSpec extends AnyFlatSpec with CromwellTimeout
...
-        override def read(cloudHost: String, cloudPath: String, offset: Long) = {
+        override def read(cloudHost: String, cloudPath: String, offset: Long): ReadableByteChannel = {
           mock[ReadableByteChannel]
         }
       }
     }
 
     // This should only be called once, not every time we ask for the channel size
-    mockSizeFunction.expects().onCall(_ => 60).once()
+    when(mockSizeFunction.apply()).thenReturn(60)
```
Comment (cjllanwarne) [inline cloud-nio/cloud-nio-impl-ftp/src/test/scala/cloud/nio/impl/ftp/FtpCloudNioFileSystemProviderSpec.scala:69]: Are the semantics of `thenReturn` the same as the `once()`?

Comment (kshakir) [reply cloud-nio/cloud-nio-impl-ftp/src/test/scala/cloud/nio/impl/ftp/FtpCloudNioFileSystemProviderSpec.scala:69]: Unfortunately, the new implicits are NOT as fancy scalamock. scalamock makes sophisticated wrappers that allow DSLs like `.once()`. At least in Scala 2.x 😉

Instead down below there's a call to:

```scala
verify(mockSizeFunction).apply()
```

That is an equivalent verification to `.once()` ensuring that `.apply()` was called once and only once on the mock.

For the lurkers, the idea behind bypassing scalamock and using the Mockito Java API directly is further discussed in the scaladocs for `trait MockSugar`.

**Review by cjllanwarne — APPROVED** [review]

**Comment by mcovarr** [comment]
Freshly merged develop [here](https://github.com/broadinstitute/cromwell/compare/ks_mc_testing_framework_tweaks) and things seemed to go pretty well. I'd still like to wait for Scala 2.13 #6724 to merge first as that's even more of a beast 🦁

**Comment by kshakir** [comment]
Conflicts resolved over in #6735

**Pull request closed without merging**